src/atc/transformers/timezone_transformer.py

```diff
@@ -0,0 +1,74 @@
+import pyspark.sql.functions as F
+from pyspark.sql import DataFrame
+from timezonefinder import TimezoneFinder
+
+from atc.etl import ExtendedTransformer
+from atc.etl.types import dataset_group
+
+
+class TimeZoneTransformer(ExtendedTransformer):
+    """
+    This transformer extracts a timezone using longitude and latitude.
+
+    Attributes:
+    ----------
+        latitude_col : str
+            name of column with latitude values
+        longitude_col : str
+            name of column with longitude values
+        dataset_input_key : str
+            input dataset identifier
+        dataset_output_key : str
+            output dataset identifier
+        column_output_name : str
+            output column name of the TimeZone column
+
+    Methods
+    -------
+    process(df: DataFrame):
+        returns the input DataFrame with a TimeZone column
+    """
+
+    def __init__(
+        self,
+        latitude_col: str,
+        longitude_col: str,
+        dataset_input_key: str = None,
+        dataset_output_key: str = None,
+        column_output_name: str = "TimeZone",
+    ):
+        super().__init__(
+            dataset_input_key=dataset_input_key, dataset_output_key=dataset_output_key
+        )
+        self.latitude_col = latitude_col
+        self.longitude_col = longitude_col
+        self.column_output_name = column_output_name
+
+    def process(self, df: DataFrame) -> DataFrame:
+        """
+        Extracts a timezone using latitude and longitude columns,
+        and outputs the input DataFrame with a TimeZone column.
+        """
+        columns = df.columns
+
+        if self.latitude_col not in columns:
+            raise Exception("The specified latitude column is not in the DataFrame")
```

Comment:
To further improve your code you could consider adding something like this to the "atc_exceptions.py" file. 

```python
class ColumnDoesNotExistException(AtcException):
    pass
```
Then import it to the file:
```python
from atc.atc_exceptions import ColumnDoesNotExistException
```

And then use it as 
```python
raise ColumnDoesNotExistException("The specified latitude column is not in the DataFrame")
```

This way it becomes easier for the uers to understand the exeptions. We can then reuse the exeption in other classes too.